lib/parser.dart

```diff
@@ -79,12 +82,12 @@ abstract class Mapping {
   /// [uri] is the optional location of the output file to find the span for
   /// to disambiguate cases where a mapping may have different mappings for
   /// different output files.
-  SourceMapSpan spanFor(int line, int column,
-      {Map<String, SourceFile> files, String uri});
+  SourceMapSpan? spanFor(int line, int column,
```

Comment:
Do you know why this has a nullable return? Can you add a TODO here to figure out why and whether we can change that?

Reply:
The `SingleMapping` class is quite permissive and returns null for a variety of reasons.

We could alternatively throw but I like that API less - it removes static errors and pushes them to runtime errors.